.github/workflows/main.yml

```diff
@@ -0,0 +1,51 @@
+name: CI
+
+on: [push, workflow_dispatch]
+
+jobs:
+  lint-backend:
+    runs-on: ubuntu-latest
+    container: node:14-alpine
+    steps:
+      - name: Checkout project
+        uses: actions/checkout@v2
+      - name: Install dependencies and lint javascript
+        run: |
+          cd backend
+          yarn install --frozen-lockfile
+          yarn lint
+  lint-frontend:
+    runs-on: ubuntu-latest
+    container: node:14-alpine
+    steps:
+      - name: Checkout project
+        uses: actions/checkout@v2
+      - name: Install dependencies and lint javascript
+        run: |
+          cd frontend
+          yarn install --frozen-lockfile
+          yarn lint
+  build-and-push-docker-image:
+    needs: [lint-backend, lint-frontend]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout project
+        uses: actions/checkout@v2
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.SPACEFILL_DOCKER_REGISTRY_USERNAME }}
```

Comment:
@Its-Alex **question:** why `username` is in a secret?

Reply:
@stephane-klein These secrets came from Github action secret directly setup in the repository

Reply:
> Do you want me to put it directly in the file?

Username is already in the repository 🤔

```
tags: spacefill/gibbon-mail:latest
```